lib/Controller/DocumentController.php

```diff
@@ -157,8 +157,21 @@ private function getDiscovery() {
 			$contact_admin = $this->l10n->t('Please contact the "%s" administrator.', [$wopiRemote]);
 
 			try {
-				$wopiClient = \OC::$server->getHTTPClientService()->newClient();
-				$discovery = $wopiClient->get($wopiDiscovery)->getBody();
+				// If we are sending query to built-in CODE server, we avoid using IClint::get() method
+				// because of an encoding issue in guzzle: https://github.com/guzzle/guzzle/issues/1758
+				if (strpos($wopiDiscovery, 'proxy.php') === false) {
+					$wopiClient = \OC::$server->getHTTPClientService()->newClient();
+					$discovery = $wopiClient->get($wopiDiscovery)->getBody();
+				}
+				else {
+					$arrContextOptions=array(
+						"ssl"=>array(
+							"verify_peer"=>false,
```

Comment:
This should not be used in a live deployment - this is a potential sec risk

Reply:
> This should not be used in a live deployment - this is a potential sec risk

Thank you for the review! :)

I have adjusted the patch to enable verify_peer, and set allow_self_signed to true. That is common for test/local/small installations.